File: app/src/main/java/com/ivancea/MTGRules/ui/main/MainViewModel.java
package com.ivancea.MTGRules.ui.main;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import java.util.Collections;
import java.util.List;

import lombok.Getter;
import lombok.NonNull;

import com.ivancea.MTGRules.model.HistoryItem;
import com.ivancea.MTGRules.model.Rule;

@Getter
public class MainViewModel extends ViewModel {
    @NonNull
    private final MutableLiveData<List<Rule>> currentRules = new MutableLiveData<>(Collections.emptyList());
    @NonNull
    private final MutableLiveData<List<Rule>> visibleRules = new MutableLiveData<>(Collections.emptyList());
    private final MutableLiveData<String> selectedRuleTitle = new MutableLiveData<>(null);
    private final MutableLiveData<String> searchText = new MutableLiveData<>(null);

    @NonNull
    private final MutableLiveData<List<HistoryItem>> history = new MutableLiveData<>(Collections.emptyList());

    @NonNull
    private final MutableLiveData<String> actionbarSubtitle = new MutableLiveData<>(null);
}

File: app/src/main/java/com/ivancea/MTGRules/ui/main/MainViewModel.kt
package com.ivancea.MTGRules.ui.main

import androidx.lifecycle.ViewModel
import com.ivancea.MTGRules.model.HistoryItem
import com.ivancea.MTGRules.model.Rule
import kotlinx.coroutines.flow.MutableStateFlow

class MainViewModel : ViewModel() {
    val currentRules = MutableStateFlow(emptyList<Rule>())
    val visibleRules = MutableStateFlow(emptyList<Rule>())
    val selectedRuleTitle = MutableStateFlow<String?>(null)
    val searchText = MutableStateFlow<String?>(null)
    val history = MutableStateFlow(emptyList<HistoryItem>())
    val actionbarSubtitle = MutableStateFlow<String?>(null)
    val showSymbols = MutableStateFlow(true)
}